### PR TITLE
LLM: update example layout

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Related to https://github.com/analytics-zoo/nano/issues/597, update LLM example layout
(Some folders are temporarily empty and will be added examples later.)

### 3. Summary of the change 

Example structure preview: https://github.com/plusbang/BigDL/tree/refactor-cpu-gpu-examples/python/llm/example
- update folder structure
- update some example links

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

**Comment by jason-dai** [comment]
1) Change Data-Type to More-Data-Types
2) Change FP-AutoModels to HF-Transformers-AutoModels
3) Add a readme in the CPU/GPU folder describing what examples each sub-folder contains

**Comment by plusbang** [comment]
> 1. Change Data-Type to More-Data-Types
> 2. Change FP-AutoModels to HF-Transformers-AutoModels
> 3. Add a readme in the CPU/GPU folder describing what examples each sub-folder contains

Have updated.

File: python/llm/example/GPU/HF-Transformers-AutoModels/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# BigDL-LLM Optimization for Large Language Model on Intel GPUs
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/README.md:1]: BigDL-LLM Examples on Intel GPU

File: python/llm/example/GPU/HF-Transformers-AutoModels/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+To help you quickly get started with transformers-style API to run any Huggingface Transformer model on Intel GPU, we provide lots of detailed examples demonstrating various usage, including:
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/README.md:3]: This folder contains examples of running BigDL-LLM on Intel GPU

File: python/llm/example/GPU/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# BigDL-LLM Optimization for Large Language Model on Intel GPUs
```
Comment (jason-dai) [inline python/llm/example/GPU/README.md:1]: BigDL-LLM Examples on Intel GPU

File: python/llm/example/GPU/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+We provide detailed examples to help you run popular open-source models using BigDL-LLM on Intel GPUs. Please refer to the appropriate guide based on different BigDL-LLM key feature:
```
Comment (jason-dai) [inline python/llm/example/GPU/README.md:3]: This folder contains examples of running BigDL-LLM on Intel GPU

File: python/llm/example/GPU/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+In general, you just need one-line `optimize_model` to easily optimize any loaded PyTorch model, regardless of the library or API you are using. See the complete example in [PyTorch-Models Folder](PyTorch-Models).
+
+## Transformers-style API
+
+Many popular open-source PyTorch large language models can be loaded using the Huggingface Transformers API (such as AutoModel, AutoModelForCasualLM, etc). For such models, BigDL-LLM also provides a set of APIs to support them. See the complete example in [HF-Transformers-AutoModels Folder](HF-Transformers-AutoModels).
+
+## QLoRA FineTuning
+
+BigDL-LLM also supports finetuning with 4bit optimizations on Intel GPUs and complete example can be found in [QLoRA-FineTuning Folder](QLoRA-FineTuning).
```
Comment (jason-dai) [inline python/llm/example/GPU/README.md:15]: remove this part

File: python/llm/example/GPU/HF-Transformers-AutoModels/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+- [Model Folder](Model): examples to run Huggingface Transformer models with INT4 optimizations.
+- [More-Data-Types Folder](More-Data-Types): examples of how to apply other low bit optimizations.
+- [Save-Load Folder](Save-Load): examples about how to save and load optimized model.
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/README.md:7]: This folder contains examples of running any Huggingface Transformer model on BigDL-LLM (using the standard AutoModel APIs)
- [Model](Model): examples of running Huggingface Transformer models (e.g., LLaMA2, ChatGLM2, Falcon, MPT, Baichuan2, etc.) using INT4 optimizations
- [More-Data-Types](More-Data-Types): examples of applying other low bit optimizations (NF4/INT5/INT8, etc.)
- [Save-Load](Save-Load): examples of saving and loading low-bit models

File: python/llm/example/GPU/PyTorch-Models/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# BigDL-LLM Optimization for Large Language Model on Intel GPUs
```
Comment (jason-dai) [inline python/llm/example/GPU/PyTorch-Models/README.md:1]: Running PyTorch model using BigDL-LLM on Intel GPU

File: python/llm/example/CPU/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# BigDL-LLM Optimization for Large Language Model on Intel CPUs
```
Comment (jason-dai) [inline python/llm/example/CPU/README.md:1]: BigDL-LLM Examples on Intel CPU

File: python/llm/example/CPU/README.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+We provide detailed examples to help you run popular open-source models using BigDL-LLM on Intel CPUs. Please refer to the appropriate guide based on different BigDL-LLM key feature:
```
Comment (jason-dai) [inline python/llm/example/CPU/README.md:3]: This folder contains examples of running BigDL-LLM on Intel CPU

**Comment by plusbang** [comment]
Have updated according to above comments.

File: python/llm/example/CPU/HF-Transformers-AutoModels/README.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Running Hugging Face Transformers model using BigDL-LLM on Intel CPU
+
+This folder contains examples of running any Hugging face Transformers model on BigDL-LLM (using the standard AutoModel APIs):
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/README.md:3]: Hugging Face

File: python/llm/example/CPU/HF-Transformers-AutoModels/README.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+- [Model](Model): examples of running Hugging face Transformers models (e.g., LLaMA2, ChatGLM2, Falcon, MPT, Baichuan2, etc.) using INT4 optimizations
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/README.md:5]: Hugging Face

File: python/llm/example/CPU/PyTorch-Models/README.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Running PyTorch model using BigDL-LLM on Intel CPU
+
+This folder contains examples of running any PyTorch model on BigDL-LLM (only need one-line code change):
```
Comment (jason-dai) [inline python/llm/example/CPU/PyTorch-Models/README.md:3]: with "one-line code change"

File: python/llm/example/GPU/PyTorch-Models/README.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# Running PyTorch model using BigDL-LLM on Intel GPU
+
+This folder contains examples of running any PyTorch model on BigDL-LLM (only need one-line code change):
```
Comment (jason-dai) [inline python/llm/example/GPU/PyTorch-Models/README.md:3]: with "one-line code change"

File: python/llm/example/GPU/HF-Transformers-AutoModels/README.md [hunk 14]
```diff
@@ -0,0 +1,7 @@
+# Running Hugging Face Transformers model using BigDL-LLM on Intel GPU
+
+This folder contains examples of running any Hugging face Transformers model on BigDL-LLM (using the standard AutoModel APIs):
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/README.md:3]: Hugging Face

File: python/llm/example/GPU/HF-Transformers-AutoModels/README.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
...
+
+- [Model](Model): examples of running Hugging face Transformers models (e.g., LLaMA2, ChatGLM2, Falcon, MPT, Baichuan2, etc.) using INT4 optimizations
```
Comment (jason-dai) [inline python/llm/example/GPU/HF-Transformers-AutoModels/README.md:5]: Hugging Face

**Comment by plusbang** [comment]
Have fixed above typos.

File: python/llm/example/GPU/README.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+# BigDL-LLM Examples on Intel GPU
```
Comment (jason-dai) [inline python/llm/example/GPU/README.md:1]: Also need to document the GPU supported and the configurations

Comment (plusbang) [reply python/llm/example/GPU/README.md:1]: > Also need to document the GPU supported and the configurations

Have added some descriptions about CPU/GPU supported in `python/llm/example/CPU/README.md` and `python/llm/example/GPU/README.md`.

File: python/llm/example/GPU/README.md [hunk 17]
```diff
@@ -5,3 +5,21 @@ This folder contains examples of running BigDL-LLM on Intel GPU:
 - [HF-Transformers-AutoModels](HF-Transformers-AutoModels): running any Hugging Face Transformers model on BigDL-LLM (using the standard AutoModel APIs)
 - [PyTorch-Models](PyTorch-Models): running any PyTorch model on BigDL-LLM (with "one-line code change")
 - [QLoRA-FineTuning](QLoRA-FineTuning): running QLoRA finetuning on BigDL-LLM
+
+
+## System Support
+**Hardware**:
+- Intel Arc™ A-Series Graphics
+- Intel Data Center GPU Flex Series
```
Comment (jason-dai) [inline python/llm/example/GPU/README.md:13]: Intel Data Center GPU Max Series

**Comment by jason-dai** [comment]
Add Max GPU; otherwise LGTM

**Comment by plusbang** [comment]
> Add Max GPU; otherwise LGTM

yeah, have added.

**Review by Oscilloscope98 — APPROVED** [review]

**Pull request merged**